google-cloud-error_reporting/lib/google/cloud/error_reporting/project.rb

```diff
@@ -0,0 +1,220 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+require "google/cloud/errors"
+require "google/cloud/env"
+require "google/cloud/error_reporting/service"
+require "google/cloud/error_reporting/credentials"
+require "google/cloud/error_reporting/error_event"
+
+module Google
+  module Cloud
+    module ErrorReporting
+      ##
+      # Project
+      #
+      # Projects are top-level containers in Google Cloud Platform. They store
+      # information about billing and authorized users, and they control access
+      # to Stackdriver ErrorReporting. Each project has a friendly name and a
+      # unique ID. Projects can be created only in the [Google Developers
+      # Console](https://console.developers.google.com).
+      #
+      # @example
+      #   require "google/cloud/error_reporting"
+      #
+      #   error_reporting = Google::Cloud::ErrorReporting.new
+      #   error_event = error_reporting.error_event "Error with Backtrace",
+      #                                             timestamp: Time.now
+      #                                             service_name: "my_app_name"
+      #   error_reporting.report error_event
+      #
+      # @see Google::Cloud::ErrorReporting.new
+      #
+      class Project
+        ##
+        # Find default project_id from ERROR_REPORTING_RPOJECT,
+        # GOOGLE_CLOUD_PROJECT, GCLOUD_PROJECT environment varaibles, or query
+        # from GCE meta service
+        #
+        # @return [String] default valid GCP project_id
+        #
+        def self.default_project
+          ENV["ERROR_REPORTING_PROJECT"] ||
+            ENV["GOOGLE_CLOUD_PROJECT"] ||
+            Google::Cloud.env.project_id
+        end
+
+        ##
+        # Find default service_name from ERROR_REPORTING_SERVICE,
+        # GAE_MODULE_NAME environment varaibles or just "ruby"
+        #
+        # @return [String] default GCP service_name
+        #
+        def self.default_service_name
+          ENV["ERROR_REPORTING_SERVICE"] ||
+            Google::Cloud.env.app_engine_service_id ||
+            "ruby"
+        end
+
+        ##
+        # Find default service_version from ERROR_REPORTING_VERSION or
+        # GAE_MODULE_VERSION environment varaibles
+        #
+        # @return [String] default GCP service_version
+        #
+        def self.default_service_version
+          ENV["ERROR_REPORTING_VERSION"] ||
+            Google::Cloud.env.app_engine_service_version
+        end
+
+        ##
+        # @private The Service object
+        attr_accessor :service
+
+        ##
+        # Create a new Project instance
+        #
+        # @param [Google::Cloud::ErrorReporting::Service] service The underneath
```

Comment:
Should this constructor be marked `@private` for docs? (I don't know this, just asking.)